tools/wptrunner/wptrunner/tests/browsers/test_base.py

```diff
@@ -0,0 +1,54 @@
+# mypy: allow-untyped-defs
+
+import sys
+from os.path import dirname, join
+from unittest import mock
+
+import pytest
+
+sys.path.insert(0, join(dirname(__file__), "..", "..", ".."))
+
+from mozlog import structuredlog
+
+from wptrunner.browsers import base
+
+
+class MozLogTestHandler(object):
+    def __init__(self):
+        self.items = []
+
+    def __call__(self, data):
+        self.items.append(data)
+
+
+@pytest.mark.parametrize("stop_browser", [False, True])
+def test_logging_immediate_exit(stop_browser):
+    logger = structuredlog.StructuredLogger("test")
+    handler = MozLogTestHandler()
+    logger.add_handler(handler)
+
+    class CustomException(Exception):
+        pass
+
+    with mock.patch.object(base, "wait_for_service", side_effect=CustomException):
+        browser = base.WebDriverBrowser(
+            logger, webdriver_binary="echo", webdriver_args=["sample output"]
```

Comment:
Does this depend on `echo` being a real binary that actually does what you expect?

Reply:
Yes, but that should be true on any POSIX system (and is true on every vaguely POSIX system I'm aware of).

Reply:
Ah, it's not true on Windows (where it's only an internal command of `cmd.exe`).

Unfortunately it's hard to partially mock `mozprocess`, so I guess let's just skip this on Windows.